implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -86,59 +149,138 @@ pub async fn print_query_status(
+            )
+            .await
+            .context("Failed to get list of tcp listeners from node")?;
+        let tcp_listeners = api::parse_tcp_list(&resp)?;
+
+        // Get list of Secure Channel Listeners
+        let resp: Vec<u8> = ctx
+            .send_and_receive_with_timeout(
+                route.clone(),
+                api::list_secure_channel_listener().to_vec()?,
+                SEND_RECEIVE_TIMEOUT_SECS,
+            )
+            .await
+            .context("Failed to get list of secure channel listeners from node")?;
+        let mut dec = Decoder::new(&resp);
+        let _ = dec.decode::<Response>()?;
+        let secure_channel_listeners = dec.decode::<Vec<String>>()?;
+
+        // Get list of inlets
+        let resp: Vec<u8> = ctx
+            .send_and_receive_with_timeout(
+                route.clone(),
+                api::list_inlets().to_vec()?,
+                SEND_RECEIVE_TIMEOUT_SECS,
+            )
+            .await
+            .context("Failed to get list of inlets from node")?;
+        let inlets = api::parse_list_inlets_response(&resp)?;
+
+        // Get list of outlets
+        let resp: Vec<u8> = ctx
+            .send_and_receive_with_timeout(
+                route.clone(),
+                api::list_outlets().to_vec()?,
+                SEND_RECEIVE_TIMEOUT_SECS,
+            )
+            .await
+            .context("Failed to get list of outlets from node")?;
+        let outlets = api::parse_list_outlets_response(&resp)?;
+
+        print_node_info(
+            &node_cfg,
+            &node_name,
+            "UP",
+            &default_id,
+            Some(&services),
+            Some(&tcp_listeners),
+            Some(&secure_channel_listeners),
+            Some((&inlets, &outlets)),
+        );
+    }
 
-    print_node_info(&node_cfg, &node_name, "UP", &default_id);
     Ok(())
 }
 
-async fn query_status(ctx: &mut ockam::Context, route: &Route) -> anyhow::Result<()> {
-    ctx.send(route.clone(), api::query_status()?).await?;
+/// Send message(s) to a node to determine if it is 'up' and
+/// responding to requests.
+///
+/// If `wait_until_ready` is `true` and the node does not
+/// appear to be 'up', retry the test at time intervals up to
+/// a maximum number of retries. A use case for this is to
+/// allow a node time to start up and become ready.
+async fn is_node_up(
+    ctx: &mut ockam::Context,
+    route: &Route,
+    wait_until_ready: bool,
+) -> anyhow::Result<bool> {
+    let mut node_up = false;
+    let attempts = match wait_until_ready {
+        true => IS_NODE_UP_ATTEMPTS,
+        false => 1,
+    };
 
-    let resp = ctx
-        .receive_duration_timeout::<Vec<u8>>(Duration::from_millis(250))
-        .await
-        .context("Failed to process request");
+    for att in 0..attempts {
+        // Sleep, if this not the first loop
+        if att > 0 {
+            tokio::time::sleep(Duration::from_millis(IS_NODE_UP_SLEEP_MILLIS)).await;
+        }
```

Comment:
Wouldn't we have the same behavior if the `sleep` is called at the end of the loop? That way we could remove the `if` block.

Reply:
I want to avoid sleeping if the caller only wants us to do one attempt.

If the `sleep` were moved to the end of the loop, I think it would still need an `if`. Consider when `wait_until_ready=false`and the node is down, without an `if` there would be a sleep before returning.

Thanks!

Reply:
Yeah, true, I didn't consider that case. Thanks for the clarification!